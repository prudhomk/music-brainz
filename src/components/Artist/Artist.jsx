import React from 'react';
import PropTypes from 'prop-types';

export default function Artist({ artist, country }) {
  // artist, genre, country, birthday
  return (
    <>
      <h1>{artist}</h1>
      <p>{country}</p>
    
            
    </>
  );
}

Artist.propTypes = {
  artist: PropTypes.string.isRequired,
  country: PropTypes.string.isRequired
};
