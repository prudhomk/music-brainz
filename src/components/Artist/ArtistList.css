.ArtistList {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.ArtistList li {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 250px;
  height: auto;
  cursor: pointer;
  list-style: none;
  margin: 5px;
  padding: 5px;
  border: 2px solid black;
  box-shadow: 2px 2px 10px 2px;
  transition: width 0.5s;
}

.ArtistList li:hover {
  background-color: pink;
  width: 300px;
}