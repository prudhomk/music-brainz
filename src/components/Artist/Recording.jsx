import React from 'react';
import PropTypes from 'prop-types';

export default function Recording({ title }) {
  return (
    <>
      <h1>{title}</h1>
    </>
  );
}

Recording.propTypes = {
  title: PropTypes.string.isRequired,
};
