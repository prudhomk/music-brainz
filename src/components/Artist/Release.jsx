import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { getArtWork } from '../../services/musicApi';

export default function Release({ id, title, art, date }) {
  const [image, setImage] = useState('');
  
  
  useEffect(async () => {
    const art = await getArtWork(id);
    console.log(art);
    await setImage(art);
 
  }, []);

  return (
    <>  
      <img src={image} alt={title} />
      <h1>{title}</h1>
      <p>{date}</p>
            
    </>
  );
}

Release.propTypes = {
  id: PropTypes.string.isRequried,
  title: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  art: PropTypes.string.isRequired
};
