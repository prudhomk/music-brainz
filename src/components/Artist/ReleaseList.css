.ReleaseList {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.ReleaseList li {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 250px;
  height: auto;
  cursor: pointer;
  list-style: none;
  margin: 5px;
  padding: 5px;
  border: 2px solid black;
  box-shadow: 2px 2px 10px 2px;
  transition: width 0.5s;
  object-fit: contain;
}

.ReleaseList img {
  object-fit: contain;
  height: 200px;
  width: 200px;
}

.ReleaseList li:hover {
  background-color: pink;
  width: 300px;
}