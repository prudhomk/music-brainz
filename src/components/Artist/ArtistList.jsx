import React from 'react';
import Artist from './Artist';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export default function ArtistList({ artist }) {
  if(artist) {
    const artistResult = artist.map((artist) => (
      <>
        <li key={artist.id}>
          <Link>
            <Artist {...artist}/>
          </Link>
        </li>;
      </>
    ));
  
    return (
      <ul>
        {artistResult} 
      </ul>
    );
  }
  return <h1>No results found</h1>;
}

ArtistList.propTypes = {
  artist: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      birthday: PropTypes.string.isRequired,
      country: PropTypes.string.isRequired,
      genre: PropTypes.string.isRequired
    })
  )
};

