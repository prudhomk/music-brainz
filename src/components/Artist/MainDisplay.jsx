import React from 'react';
import { useState, useEffect } from 'react';
// import useArtist from '../state/useArtist';
import { artistSearch, mungeArtist } from '../../services/musicApi';
import ArtistList from '../Artist/ArtistList';

export default function MainDisplay() {
  const [offSet, setOffSet] = useState(0);
  const [page, setPage] = useState(1);
  const [artist, setArtist] = useState('');
  const [loading, setLoading] = useState(true);
  const [artistList, setArtistList] = useState([]);

  useEffect(() => {
    setLoading(false);
    return () => {
      return;
    };
  }, [handleSubmit]
  );
  
  const handleChange = ({ target }) => {
    setArtist(target.value);
  };

  const handleIncrement = async (e) => {
    setPage((prevPage) => prevPage + 1);
    setOffSet((prevOffSet) => prevOffSet + 5);
    const query = await artistSearch(artist, offSet);

    const searchedArtist = await mungeArtist(query);
    setArtistList(searchedArtist);
  };

  const handleDecrement = async (e) => {
    setPage((prevPage) => prevPage - 1);
    setOffSet((prevOffSet) => prevOffSet - 5);
    const query = await artistSearch(artist, offSet);

    const searchedArtist = await mungeArtist(query);
    setArtistList(searchedArtist);
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    const query = await artistSearch(artist, offSet);

    const searchedArtist = await mungeArtist(query);
    setArtistList(searchedArtist);
  };

 
  return (
    <div data-testid="display">
      <form onSubmit={handleSubmit}>
    Artist: <input type="text" placeholder="Artist" name="searchTerm" onChange={handleChange}/>
        <button data-testid="submitButton">SEARCH</button>
      </form>
    
      <button
        data-testid="decrement"
        disabled={page <= 1}
        onClick={handleDecrement}>
                -
      </button> 
      <span data-testid="page">Page: {page}</span>
      <button 
        data-testid="increment"
        onClick={handleIncrement}>
      +
      </button>
      <ArtistList artistList={artistList}/>
    </div>
  );
}
