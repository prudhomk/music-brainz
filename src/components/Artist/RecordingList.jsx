import React, { useState, useEffect } from 'react';
import Recording from './Recording';
import { Link, useParams } from 'react-router-dom';
import PropTypes from 'prop-types';
import { mungeRecordings, getRecordings } from '../../services/musicApi';


export default function RecordingList() {
  const [recordings, setRecordings] = useState([]);
  const [loading, setLoading] = useState(true);
  const { id } = useParams();

  useEffect(async () => {
    const recordings = await getRecordings(id);
   
    const mungedRecordings = await mungeRecordings(recordings);
    console.log(mungedRecordings);
    await setRecordings(mungedRecordings);
    
    setLoading(false);
  }, []);

  if(recordings) {
    const recordingResult = recordings.map((recording) => (
      <>
        <li key={recording.id}>
          <Link to={`/${recording.id}/lyrics`}>
            <Recording {...recording} />
          </Link>
        </li>
      </>
    
    ));

    return (
      <>
        <h1>Recording List</h1>
        <ul>
          {recordingResult}
        </ul>
      </>
    );
  }


  return (
    <div>
      placeholder
    </div>
  );
}

RecordingList.propTypes = {
  recordingList: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
    })
  )
};

