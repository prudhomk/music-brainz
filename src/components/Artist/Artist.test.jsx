import React from 'react';
import { render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import Artist from './Artist';

describe('Main display component snapshot', () => {
  it('render Artist', () => {
    const artist = {
      artist: 'name',
      country: 'country'
    };

    act(() => {
      const { asFragment } = render(
        <Artist {...artist}/>
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
