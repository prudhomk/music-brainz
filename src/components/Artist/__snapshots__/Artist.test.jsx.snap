// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main display component snapshot render Artist 1`] = `
<DocumentFragment>
  <h1>
    name
  </h1>
  <p>
    country
  </p>
</DocumentFragment>
`;
